java/junit5/src/test/java/com/saucelabs/saucebindings/junit5/CapabilitiesTest.java

```diff
@@ -21,16 +24,17 @@ public static Capabilities getCapabilities() {
   }
 
   @RegisterExtension
-  static SauceBindingsExtension sauceExtension = new SauceBindingsExtension(getCapabilities());
+  static SauceBindingsExtension sauceExtension =
+      new SauceBindingsExtension.Builder().withCapabilities(getCapabilities()).build();
 
   @Test
-  public void useCustomCapabilities() {
-    RemoteWebDriver driver = (RemoteWebDriver) sauceExtension.getDriver();
-    Assertions.assertEquals("Safari", driver.getCapabilities().getBrowserName());
+  public void useCustomCapabilities(SauceSession session, WebDriver driver) {
```

Comment:
Yes, but it can be in a before hook in their base class that just stores the variables. The examples are skipping that for simplicity right now.

This implementation is great because you an do either or both in either order or none.
I originally had it only for Session, and put in code for user to call `session.getDriver()`. But I think it is simpler to show it with both.

It's just adding a conditional to 2 methods to allow:
```
@BeforeEach
public void setup(SauceSession session, Playwright playwright)
```

Reply:
Requiring the user to change their code for every test might make people steer away.

Reply:
Would it be less confusing to always use a before hook?

Like it's obvious that the difference between what we had previously vs now is simple:
```
@BeforeEach
public void setUp() {
  this.session = extension.getSession();
  this.driver = extension.getDriver();
}
```

Now:
```
@BeforeEach
public void setUp(SauceSession session, WebDriver driver) {
  this.session = session;
  this.driver = driver;
}
```

Maybe I'm being too cute by removing the before hook in the examples just for fewer lines of code.

Reply:
Oh no, I did the right thing. The _examples_ all show it in a before hook for clarity. In the _tests_ I'm using a shortcut because it's the tests.